lib/active_record/tenanted/database_adapters/mysql.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module Tenanted
+    module DatabaseAdapters # :nodoc:
+      class MySQL
+        attr_reader :db_config
+
+        def initialize(db_config)
+          @db_config = db_config
+        end
+
+        def tenant_databases
+          like_pattern = db_config.database_for("%")
+          scanner_pattern = db_config.database_for("(.+)")
+          scanner = Regexp.new("^" + Regexp.escape(scanner_pattern).gsub(Regexp.escape("(.+)"), "(.+)") + "$")
+
+          begin
+            ActiveRecord::Tasks::DatabaseTasks.with_temporary_connection(configuration_hash_without_database) do |connection|
```

Comment:
Maybe the idiom here should be to wrap up the anonymous configuration with the connection creation?

```ruby
def with_anonymous_connection(&block)
  config_hash = configuration_hash_without_database
  ActiveRecord::Tasks::DatabaseTasks.with_temporary_connection(config_hash, &block)
end
```